### PR TITLE
Implement replication functionality on mock node

### PR DESCRIPTION
**Pull request by sandreae**
- `replicate` method for mock nodes to exchange entries
- ~functionality to parse an array of operations into a `mermaid-js` string~

## 📋 Checklist

- [x] Add tests that cover your changes
- [x] Add this PR to the _Unreleased_ section in `CHANGELOG.md`
- [x] Link this PR to any issues it closes
- [x] New files contain a SPDX license header
- [x] Check if descriptions and terminology match `handbook` content (and visa-versa)

### CODE REVIEW

**Comment by sandreae** [comment]
Yeh, that's a good point. For now though, rather than splitting off, I might prefer to implement all the mermaid-js stuff directly in `panda-playground` instead of here in `test_utils`. That would mean the only addition here is a few methods on the mock node. Not against splitting it out, would want to think more about it though.

File: p2panda-rs/src/test_utils/mocks/node.rs [hunk 1]
```diff
@@ -200,6 +200,57 @@ impl Node {
...
+                &entry.entry_encoded(),
+                &entry.operation_encoded(),
+            );
+            self.db.insert(author.as_str().into(), author_logs);
+            true
+        }
+    }
+
+    /// Replicate entries from other_node to this node.
+    pub fn replicate_with(&mut self, other_node: &mut Node) -> Vec<Hash> {
```
Comment (cafca) [inline p2panda-rs/src/test_utils/mocks/node.rs:240]: Could be nice to have examples for mock node methods. I know they are also in the tests so not required, just - nice to have :-)

**Review by cafca — APPROVED** [review]

**Review by cafca — CHANGES_REQUESTED** [review]
Actually :D It would be nice to cover all new lines with tests, I think not much is needed for that.

**Comment by sandreae** [comment]
I'm gunna close this as most of the functionality was moved out into the playground, and soon we will do a revamp of the mock node in any case. This can be approach then if it's still needed.

**Pull request closed without merging**